scripts/vault_login.py

```diff
@@ -0,0 +1,100 @@
+    "install vault CLI as per your system spec!"
+)
+
+
+def _load_env():
+    print("\nFetching vault data from .env file and Exporting vault URL .....\n")
+    ROOT_PATH = os.path.realpath(
+        os.path.join(os.path.dirname(os.path.abspath(__file__)), os.path.pardir)
+    )
+    with open(ROOT_PATH + '/.env') as envfile:
+        envdata = envfile.read()
+    return ROOT_PATH, envdata
+
+
+def _export_vault_url(envdata):
+    vaulturl = re.findall('VAULT_URL_FOR_DYNACONF=(.*)', envdata)[0]
+    if 'localhost:8200' in vaulturl:
+        raise InvalidVaultURLForOIDC(
+            f"\033[3;31m{vaulturl} doesnt supports OIDC login,"
+            "please change url to corp vault in env file!"
+        )
+    os.environ['VAULT_ADDR'] = vaulturl
+
+
+def _vault_command(command: str):
+    vcommand = subprocess.run(command, capture_output=True, shell=True)
+    if vcommand.returncode != 0:
+        verror = str(vcommand.stderr)
+        if 'vault: command not found' in verror:
+            print(f"\033[1;31mError! {HELP_TEXT}")
+        elif 'Error revoking token' in verror:
+            print("\033[1;32mToken is alredy revoked!")
+        elif 'Error looking up token' in verror:
+            print("\033[1;33mWarning! Vault not logged in, please run 'make vault-login'!")
+        else:
+            print(f"\033[1;31mError! {verror}")
+        sys.exit(0)
+    return vcommand
+
+
+def _vault_login(root_path, envdata):
+    print(
+        "\033[1;30mWarning! The browser is about to open for vault OIDC login, "
+        "close the tab once the sign in is done!"
+    )
+    time.sleep(5)
+    if _vault_command(command="vault login -method=oidc").returncode == 0:
+        os.system("vault token renew -i 10h")
+        print("\033[1;32mSuccess! Vault OIDC Logged-In and extended for 10 hours!")
+    # Fetching token
+    token = _vault_command("vault token lookup --format json").stdout
+    token = json.loads(str(token.decode('UTF-8')))['data']['id']
+    # Setting new token in env file
+    envdata = re.sub('.*VAULT_TOKEN_FOR_DYNACONF=.*', f"VAULT_TOKEN_FOR_DYNACONF={token}", envdata)
+    with open(root_path + '/.env', 'w') as envfile:
+        envfile.write(envdata)
+    print("\033[1;32mSuccess! New OIDC token is added to .env file to access secrets from vault!")
+
+
+def _vault_logout(root_path, envdata):
+    # Teardown - Setting dymmy token in env file
+    envdata = re.sub('.*VAULT_TOKEN_FOR_DYNACONF=.*', "# VAULT_TOKEN_FOR_DYNACONF=myroot", envdata)
+    with open(root_path + '/.env', 'w') as envfile:
+        envfile.write(envdata)
+    vlogout = _vault_command('vault token revoke -self')
+    if vlogout.returncode == 0:
+        print("\033[1;32mSuccess! OIDC token removed from Env file successfully!")
+
+
+def _vault_status():
+    vstatus = _vault_command('vault token lookup')
+    if vstatus.returncode == 0:
+        print(str(vstatus.stdout.decode('UTF-8')))
+
+
+if __name__ == '__main__':
```

Comment:
I feel like this is written in bash, in python :) 

Click is really good for creating extensible command structures. This file is largely a convenience - the only variable necessary to be able to use vault is the token, and to be able to use `vault` on the command line the URL.

I don't want us mirroring the vault CLI instead of using both `VAULT_ADDR` and `VAULT_URL_FOR_DYNACONF` in the `.env.example` file. 

That file is read by dynaconf, but it also can be sourced in one's local env in order to just use `vault` CLI directly and have the URL set.

Having a tool that makes integrating with the vault CLI is helpful, but I'm not sure this is the right approach. Using python's entrypoints and command groups implemented through click creates a far more extensible interface for doing things like this to 'work' with the framework and setup a testing environment.

Reply:
Ofcorse there are multiple ways of implementing things, but since this is just a helper module to be used with `Make`, I kept it simple! 

I would not clock the PR on this helper module improvements/re-implementation, we can merge this PR and later one can raise a PR to improve it !